python/lsst/meas/photocal/PhotoCal.py

```diff
@@ -401,64 +414,68 @@ def extractMagArrays(self, matches, filterName, keys):
             ct = None
 
         if ct:                          # we have a colour term to worry about
-            fluxNames = [ct.primary, ct.secondary]
-            missingFluxes = []
-            for flux in fluxNames:
+            fluxFieldList = [getRefFluxField(refSchema, filt) for filt in (ct.primary, ct.secondary)]
+            missingFluxFieldList = []
+            for fluxField in fluxFieldList:
                 try:
-                    refSchema.find(flux).key
+                    refSchema.find(fluxField).key
                 except KeyError:
-                    missingFluxes.append(flux)
+                    missingFluxFieldList.append(fluxField)
 
-            if missingFluxes:
+            if missingFluxFieldList:
                 self.log.warn("Source catalog does not have fluxes for %s; ignoring color terms" %
-                              " ".join(missingFluxes))
+                              " ".join(missingFluxFieldList))
                 ct = None
 
         if not ct:
-            fluxNames = ["flux"]
+            fluxFieldList = [getRefFluxField(refSchema, filterName)]
 
-        refFluxes = []
-        refFluxErrors = []
-        for flux in fluxNames:
-            fluxKey = refSchema.find(flux).key
-            refFlux = np.array([m.first.get(fluxKey) for m in matches])
+        refFluxArrList = [] # list of ref arrays, one per flux field
+        refFluxErrArrList = [] # list of ref flux arrays, one per flux field
+        for fluxField in fluxFieldList:
+            fluxKey = refSchema.find(fluxField).key
+            refFluxArr = np.array([m.first.get(fluxKey) for m in matches])
             try:
-                fluxErrKey = refSchema.find(flux + ".err").key
-                refFluxErr = np.array([m.first.get(fluxErrKey) for m in matches])
+                fluxErrKey = refSchema.find(fluxField + "Sigma").key
+                refFluxErrArr = np.array([m.first.get(fluxErrKey) for m in matches])
             except KeyError:
-                # Catalogue may not have flux uncertainties; HACK
+                # Reference catalogue may not have flux uncertainties; HACK
                 self.log.warn("Reference catalog does not have flux uncertainties for %s; using sqrt(flux)."
-                              % flux)
-                refFluxErr = np.sqrt(refFlux)
+                              % fluxField)
+                refFluxErrArr = np.sqrt(refFluxArr)
```

Comment:
I have opened DM-2308 about this hack and the later hack of scaling reference flux errors. As Jim notes in the later comment it may be more robust to ignore reference flux error entirely, so I propose to adopt that solution until DM-2308 is properly dealt with.

On Mar 11, 2015, at 12:08 PM, Jim Bosch <notifications@github.com> wrote:

> In python/lsst/meas/photocal/PhotoCal.py:
> 
> >                  self.log.warn("Reference catalog does not have flux uncertainties for %s; using sqrt(flux)."
> > -                              % flux)
> > -                refFluxErr = np.sqrt(refFlux)
> > +                              % fluxField)
> > +                refFluxErrArr = np.sqrt(refFluxArr)
> I don't particularly like this default (and apparently neither did the original author, hence the "HACK"). Mind opening a new issue for this?
> 
> —
> Reply to this email directly or view it on GitHub.
>